src/maliput/routing/route.cc

```diff
@@ -360,5 +365,130 @@ int Route::FindDirectionTowardsLaneSRangeWithStraightPredecessor(const Route::La
   return kTowardsRight;
 }
 
+namespace {
+
+// Finds a RoadPosition by matching @p lane pointer.
+// @param lane Pointer to the api::Lane to match.
+// @param positions Vector or positions to look for one with a matching api::Lane pointer with @p lane.
+// @return An optional with a copy of the first position found with matching api::Lane pointer with @p lane.
+std::optional<api::RoadPosition> FindPositionByLane(const api::Lane* lane,
+                                                    const std::vector<api::RoadPosition>& positions) {
+  const auto it = std::find_if(positions.begin(), positions.end(),
+                               [lane](const api::RoadPosition& position) { return position.lane == lane; });
+  return it != positions.end() ? std::optional<api::RoadPosition>{*it} : std::nullopt;
+}
+
+// Evaluates whether @p pos_a is close to @p pos_b with @p tolerance in the INERTIAL-Frame.
+// @param pos_a Left hand side of the comparison.
+// @param pos_b Right hand side of the comparison.
+// @param tolerance Tolerance to compare the positions.
+// @return true When @p pos_a and @p pos_b are within @p tolerance in the INERTIAL-Frame.
+bool IsRoadPositionClose(const api::RoadPosition& pos_a, const api::RoadPosition& pos_b, double tolerance) {
+  const api::InertialPosition& inertial_pos_a = pos_a.lane->ToInertialPosition(pos_a.pos);
+  const api::InertialPosition& inertial_pos_b = pos_b.lane->ToInertialPosition(pos_b.pos);
+  const common::ComparisonResult<api::InertialPosition> comparison_result =
+      api::IsInertialPositionClose(inertial_pos_a, inertial_pos_b, tolerance);
+  return !comparison_result.message.has_value();
+}
+
+// Returns the closes api::LaneEnd::Which of the api::Lane that @p position is in.
+api::LaneEnd::Which GetClosestLaneEndWhich(const api::RoadPosition& position) {
+  const api::Lane* lane = position.lane;
+  return position.pos.s() <= lane->length() - position.pos.s() ? api::LaneEnd::Which::kStart
+                                                               : api::LaneEnd::Which::kFinish;
+}
+
+// Matches @p position_a in the preceeding Phase with an api::RoadPosition in @p positions_b from
```

Comment:
Typo: "preceeding"

(Ditto other instances below.)

Reply:
[5f99c1c](https://github.com/maliput/maliput/pull/628/commits/5f99c1c613ff4ae3206bf0796b0287698f0d08ae)

I'll create a follow up PR to fix code.